### ISSUE TITLE
storing raw JSON values

### ISSUE THREAD

**Issue by TheRealSeber** [opral/inlang-paraglide-js#248]
Hey,

I am just curious if such functionalty as [this](https://github.com/kaisermann/svelte-i18n/blob/main/docs/Formatting.md#jsonmessageid-string) exists in the paraglide-js

I would be interested in storing JSON structures which change dependingly on the language of course. Example use case:
```svelte
{#each features as feature}
	<div class="relative px-6">
		<h1
			class="flex flex-col items-center text-center text-xl font-semibold leading-7 text-neutral-700"
		>
			<Svg name={feature.iconName} />
			{feature.title}
		</h1>
		<dd class="m-2 text-center text-lg leading-7 text-neutral-600">
			{feature.description}
		</dd>
	</div>
{/each}
```

Assuming I have `10` or more features seems to be pointless for me to store every `feature` as standalone string

**Comment by samuelstroschein** [comment]
No, such functionality doesn't exist in Paraglide JS but isn't the `$json` function just `JSON.parse()`? 

1. Store the JSON as a string in a message
2. Call `JSON.parse(message)` 

_Example message that stores a JSON as string_

```
blue_moon = '[{ iconName, title, description }]'
```

_Calling the message and parse it as json 

```ts
const features = JSON.parse(m.blue_moon())

for (const feature of features) {
  feature.iconName
  feature.title
  // ...
}

```

**Comment by TheRealSeber** [comment]
Oh well, if that's just gonna work, then that's fair enough. However I think that it would be `nice to have` thing as storing json values as strings is not pretty. Idk tho how everything works under the hood, so can't tell if it's even possible for you to add it.

**Comment by samuelstroschein** [comment]
A utility function could surely be added, but I am wary of implications in an end-to-end localization process.

- how will translators interact with json? 
- how will designers interact with json?  

If the workaround works, can you close the issue for now?

**Comment by TheRealSeber** [comment]
Reopening the issue as the workaround doesn't work @samuelstroschein 

e.g. message snippet I used: `"landing_page_features": "[{\"title\":\"Reliability and Security\",\"iconName\":\"shieldIcon\",\"description\":\"We prioritize (...)`

Output function of paraglide js after compilance:
![image](https://github.com/user-attachments/assets/48e887fe-4306-41ce-94f0-b23c8ec3d69b)

It makes sence as paraglide expects for the data in `{}` to be a parameters

**Comment by samuelstroschein** [comment]
Yep certainly so. Good catch. 

Small clarification: Paraglide JS compiles what a plugin imported. Hence, it's the [message format plugin](https://inlang.com/m/reootnfj/plugin-inlang-messageFormat) which interprets `{}` as variable expressions. Expressions that Paraglide compiles. 

I recommend to split the json properties into messages e.g. `landing_page_feature_title`, `landing_page_feature_icon_name`, etc. 

Updating the message format plugin or Paraglide JS does not make sense atm as both are on a new major version in the dev branch [waiting for lix 1.0 to be released](https://opral.substack.com/p/focus-shift-from-inlang-to-lix).
